crates/solver/src/settlement_ranker.rs

```diff
@@ -164,11 +164,33 @@ impl SettlementRanker {
             );
         }
 
-        let (mut rated_settlements, errors) = self
+        let (rated_settlements, errors) = self
             .settlement_rater
             .rate_settlements(solver_settlements, external_prices, gas_price)
             .await?;
 
+        // Discard solutions with negative obj value
+        let mut rated_settlements = rated_settlements
+            .into_iter()
+            .filter(|(solver, rated_settlement, _)| {
```

Comment:
`.into_iter().filter().collect()` could be replaced with `.retain()`.

Reply:
Ah yes. Thanks